src/CoreBundle/private/library/classes/TCMSResourceCollection.class.php

```diff
@@ -734,4 +809,18 @@ protected function isLocalResource($sResource)
 
         return $bIsLocal;
     }
+
+    private function resourceCollectionProcessRunning(): bool
+    {
+        return file_exists($this->assetPath.'/lock.tmp');
```

Comment:
Maybe (additionally) use something quicker than this? (memcache/redis/db)?

And/or check for the existence of a result (ressource collection of another process) before overwriting?

Reply:
i need this to be on a per code basis, and i need this to work even if caching is not enabled. So a file_exists looks like a reasonable solution.

As for checking if the files exists - we do that in src/CoreBundle/private/library/classes/TCMSResourceCollection.class.php lines 202-205